lib/util/lib/platform.c

```diff
@@ -217,9 +262,6 @@ hse_platform_init(void)
     if (err)
         goto errout;
 
-    dt_init();
-    event_counter_init();
-
```

Comment:
The order here used to be extremely important, perhaps that's changed, now I have to go review these functions...

Reply:
They need to be first because we apparently can't trust ourselves not to use ev() prior to initializing the data tree and event counters.

Reply:
I fixed that a long long time ago.  We should be able to call event_counter() before dt_init() and event_counter_init(), and looks like we still can.  It just puts the counter on the pending list, all the locks and stuff are initialized by the c-runtime.  Same for dt_init().  First caller to dt_find() processes the list.

Reply:
I can change it back, but that caused undefined behavior according to clang. There is probably a better explanation, but this was the easiest solution

Reply:
How/where are you running clang to produce all these issues?  I am not seeing them.  All all deltas in this commit fixes for clang detected problems?